packages/vitest/src/runtime/runners/test.ts

```diff
@@ -27,15 +27,15 @@ export class VitestTestRunner implements VitestRunner {
     this.snapshotClient.clear()
   }
 
-  async onAfterRunFiles() {
-    const result = await this.snapshotClient.finishCurrentRun()
-    if (result)
-      await rpc().snapshotSaved(result)
-  }
-
-  onAfterRunSuite(suite: Suite) {
+  async onAfterRunSuite(suite: Suite) {
     if (this.config.logHeapUsage && typeof process !== 'undefined')
       suite.result!.heap = process.memoryUsage().heapUsed
+
+    if (suite.mode !== 'skip' && typeof suite.filepath !== 'undefined') {
+      const result = await this.snapshotClient.finishCurrentRun()
+      if (result)
+        await rpc().snapshotSaved(result)
+    }
```

Comment:
I moved `finishCurrentRun` from `onAfterRunFiles` hook to `onAfterRunSuite` hook as a refactoring since it would match the logic of calling `startCurrentRun` in `onBeforeRunSuite`.